vochain/scrutinizer/process.go

```diff
@@ -1,87 +1,139 @@
 package scrutinizer
 
 import (
-	"bytes"
 	"fmt"
 	"math/big"
-	"sync/atomic"
+	"time"
 
-	"github.com/dgraph-io/badger/v2"
+	"github.com/timshannon/badgerhold/v3"
 	"go.vocdoni.io/proto/build/go/models"
-	"google.golang.org/protobuf/proto"
 
 	"go.vocdoni.io/dvote/log"
-	"go.vocdoni.io/dvote/types"
-	"go.vocdoni.io/dvote/util"
 )
 
 // ProcessInfo returns the available information regarding an election process id
-func (s *Scrutinizer) ProcessInfo(pid []byte) (*models.Process, error) {
-	return s.VochainState.Process(pid, false)
+func (s *Scrutinizer) ProcessInfo(pid []byte) (*Process, error) {
+	proc := &Process{}
+	err := s.db.FindOne(proc, badgerhold.Where("ID").Eq(pid))
+	return proc, err
 }
 
 // ProcessList returns the list of processes (finished or not) for a specific entity.
-func (s *Scrutinizer) ProcessList(entityID []byte, fromID []byte, max int64) ([][]byte, error) {
-	plistkey := []byte{types.ScrutinizerEntityPrefix}
-	plistkey = append(plistkey, entityID...)
-	processList, err := s.Storage.Get(plistkey)
-	if err != nil {
-		return nil, err
-	}
-	processListResult := [][]byte{}
-	fromLock := len(fromID) > 0
-	for _, process := range util.SplitBytes(processList, types.ProcessIDsize) {
-		if max < 1 || len(process) < 1 {
-			break
-		}
-		if !fromLock {
-			keyCopy := make([]byte, len(process))
-			copy(keyCopy, process)
-			processListResult = append(processListResult, keyCopy)
-			max--
-		}
-		if fromLock && bytes.Equal(fromID, process) {
-			fromLock = false
-		}
+func (s *Scrutinizer) ProcessList(
+	entityID []byte,
+	namespace uint32,
+	from,
+	max int) ([][]byte, error) {
```

Comment:
any particular reason you write func definitions like this now? it almost seems like you're trying to stick to 40 columns or something. I personally don't find it particularly easy to read, and it takes a ton of vertical space :)

Reply:
if this is because of the line length linter, that's part of why I think it's a bad linter. it enforces strict line length limits, which are often a bad idea.

it's true that a func decl taking 140 columns is generally bad, but then you should just split that line into two of ~70 columns. A function definition that takes 60, 80, or even 100 columns in a single line is perfectly fine and should not be split.

Reply:
Yes, the reason is because a limit of 111 characters per line is enforced by the CI and I think that's a good number to respect. Gofumt (you recommended it) also likes this way of writing function headers btw.

I can write it with less columns per function definition (splitting in two or three columns max). However that's a personal style preference, AFAIK there is nothing officially written regarding this matter.